core/src/test/java/hudson/security/HudsonPrivateSecurityRealmTest.java

```diff
@@ -0,0 +1,134 @@
+package hudson.security;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.mockStatic;
+import static org.mockito.Mockito.when;
+
+import hudson.security.HudsonPrivateSecurityRealm.JBCryptEncoder;
+import hudson.security.HudsonPrivateSecurityRealm.PBKDF2PasswordEncoder;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.time.Duration;
+import javax.crypto.SecretKeyFactory;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+
+public class HudsonPrivateSecurityRealmTest {
+
+    // MySecurePassword
+    private static final String PBKDF2_HMAC_SHA512_ENCODED_PASSWORD =
+            "$HMACSHA512:210000:30f9e0a5470a8bc67f128ca1aae25dd4$88abaca4f442caeff0096ec0f75df2d77cc31a956c564133232f4d2532a72c8d4380a718d5b2a3dccab9e752027eeadd8f9f2c0c624505531bf3a57ec7d08aad";
+
+    //@Test
```

Comment:
Purpose?

Reply:
incase people want to check the complexity of how long this takes (ie is the number of itterations we use correct for the state of CPUs).

poor mans benchmark that is not excersized but that someone can run pretty quickly if they wanted to check.

I *could* make this a test by asserting the test took > X to run - but that seemed like a test that would just waste time.

Reply:
https://github.com/jenkinsci/jenkins/blob/f37bcc31de0a041a2f1782fa3bf253701d04ebcd/core/src/main/java/hudson/security/HudsonPrivateSecurityRealm.java#L950 FTR

Reply:
resolved in 15e6c88be791c73f1e4a4b4afeabac0473ec7ce9

Reply:
> FTR

FWIW I plan to submit https://github.com/daniel-beck/jenkins/commit/93718b019ec17c859fbef4fe009a671875bf78d1 once this PR is in (unnecessary merge conflict)